beacon-light-client/solidity/contracts/validators_accumulator/ValidatorsAccumulator.sol

```diff
@@ -81,28 +80,31 @@ contract ValidatorsAccumulator is IValidatorsAccumulator {
       size /= 2;
     }
 
-    snapshots[validatorsCount - 1] = DepositData({
-      blockNumber: block.number,
-      accumulator: _getRoot(validatorsCount)
-    });
+    uint64 blockNumber = uint64(block.number);
+    snapshots[blockNumber] = _getRoot(validatorsCount);
+    uint256 blockNumbersLength = blockNumbers.length;
+    if (
+      blockNumbersLength == 0 ||
+      blockNumbers[blockNumbersLength - 1] != blockNumber
+    ) {
+      blockNumbers.push(blockNumber);
+    }
```

Comment:
I think it would be better if we used a ring buffer

Reply:
We have discussed the idea of the ring buffer but because it is a very permissionless protocol we decided that someone could make multiple deposits and delete old data, maybe this assumption is extremely pessimistic as those deposits are not cheap and we learned later the contract is not entirely permissionless. 
But apart from the ring buffer, I like the idea of using an array actually instead of a map I think it will make the solution a bit cleaner
```
DepositData[] snapshots;
```
And if the deposit is for the same block we will just overwrite the deposit

Reply:
Using mapping + array costs less in both `deposit` and `verify` transactions. Therefore, we will stick to the current approach.